packages/codec/lib/compilations/utils.ts

```diff
@@ -42,12 +56,16 @@ export function shimArtifacts(
 
 /**
  * shims a bunch of contracts ("artifacts", though not necessarily)
- * to a compilation.  usually used via one of the above two functions.
+ * to a compilation.  usually used via one of the above functions.
+ * Note: if you pass in inputSources, sources will not have
+ * compiler set, so you should set that up separately, as in
+ * shimCompilation().
  */
 export function shimContracts(
   artifacts: (Artifact | CompiledContract)[],
   files?: string[],
-  shimmedCompilationId: string = "shimmedcompilation"
+  shimmedCompilationId: string = "shimmedcompilation",
+  inputSources?: CompilerSource[]
```

Comment:
So is ≤one of `inputSources`/`files` used? Should we make the interface reflect this?

Reply:
Yeah, if `inputSources` is given then `files` is ignored.  Which now that you mention it does make for an awkward interface, yes.  Hm.  Not sure what to do here.  Any ideas?

In the worst case I can just add a comment, but that's obviously not ideal...

Reply:
Well, you could make it `options: { files, etc } | { inputSources, etc }`

Reply:
Yeah, I was hoping to avoid a breaking change.  Although IDK, this method, while exported, is pretty internal... maybe it's OK if we break it?  Or even unexport it? :P  If that's OK then I could totally do it that way.  Or we can just accept another breaking change to codec. :P

Reply:
OK, seems we're breaking it!